nemo/collections/nlp/modules/common/text_generation_utils.py

```diff
@@ -876,6 +879,7 @@ def tab_sample_sequence_batch(
                 prev = torch.clamp(prev, max=tokenizer.vocab_size - 1)
 
                 new_tokens = switch(tokens[:, context_length].view(-1), prev, started)
+                new_tokens = switch(new_tokens, eod_id, is_done)
```

Comment:
For tabular data generation, still preserve the old behavior.

Reply:
sure thing